test/integration/workflows/DerivativeIntegration.t.sol

```diff
@@ -0,0 +1,377 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.26;
+
+// external
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { ICoreMetadataModule } from "@storyprotocol/core/interfaces/modules/metadata/ICoreMetadataModule.sol";
+import { IIPAccount } from "@storyprotocol/core/interfaces/IIPAccount.sol";
+import { ILicensingModule } from "@storyprotocol/core/interfaces/modules/licensing/ILicensingModule.sol";
+import { PILFlavors } from "@storyprotocol/core/lib/PILFlavors.sol";
+
+// contracts
+import { ISPGNFT } from "../../../contracts/interfaces/ISPGNFT.sol";
+import { WorkflowStructs } from "../../../contracts/lib/WorkflowStructs.sol";
+
+// test
+import { BaseIntegration } from "../BaseIntegration.t.sol";
+
+contract DerivativeIntegration is BaseIntegration {
+    using Strings for uint256;
+
+    ISPGNFT private spgNftContract;
+    address[] private parentIpIds;
+    uint256[] private parentLicenseTermIds;
+    address private parentLicenseTemplate;
+
+    /// @dev To use, run the following command:
+    /// forge script test/integration/workflows/DerivativeIntegration.t.sol:DerivativeIntegration \
+    /// --rpc-url=$TESTNET_URL -vvvv --broadcast --priority-gas-price=1 --legacy
+    function run() public override {
+        super.run();
+        _beginBroadcast();
+        _logTestStart("DerivativeIntegration");
+        _setUpTest();
+        _test_mintAndRegisterIpAndMakeDerivative();
+        _test_registerIpAndMakeDerivative();
+        _test_mintAndRegisterIpAndMakeDerivativeWithLicenseTokens();
+        _test_registerIpAndMakeDerivativeWithLicenseTokens();
+        _test_multicall_mintAndRegisterIpAndMakeDerivative();
+        _logTestEnd("DerivativeIntegration");
+        _endBroadcast();
+    }
+
+    function _test_mintAndRegisterIpAndMakeDerivative() private {
```

Comment:
It seems not easy to determine which test failed when failure happen. might consider add modifier to each function, to log test start and end.

Reply:
Yeah good point! Will add this improvement in another PR 🫡